### ISSUE TITLE
EM on multiple sequences of observed states

### ISSUE THREAD

**Issue by lucach** [regevs/factorial_hmm#1]
I would be nice to have the `EM` method of `FullDiscreteFactorialHMM` also support multiple sequences of observed states.
For instance, the `fit` method of `hmmlearn` allows parameters estimation from multiple sequences for basic (i.e., non-factorial) HMMs (see https://github.com/hmmlearn/hmmlearn/blob/master/lib/hmmlearn/base.py#L440-L488).

This probably requires to split `MStep` in two parts: the former (equivalent of hmmlearn's `_accumulate_sufficient_statistics`) will collect statistics about the parameters and will need to be called on every sequence of observed states; the latter (equivalent of hmmlearn's  `_do_mstep`) will be required to normalize the parameters and will need to be called only once per iteration.

**Comment by regevs** [comment]
Thanks for your suggestion. The FullDiscreteFactorialHMM is actually not
useful for most practical purposes and exists mainly as a simple example.
In almost all use cases, one would implement an HMM where the
parametrization is not completely general. In such cases, it would be
straightforward to use the EStep and MStep for multiple sequences as you
described.

**Comment by lucach** [comment]
Sure, I was just pointing out that, for instance, the `MStep` in the current form does not directly allow to use multiple sequences. I've used your library in an academic toy example, making the necessary modifications to allow it (see https://github.com/lucach/fhmm_bach/blob/master/factorial_hmm_lib.py#L568-L694).

(Unrelated: I think you may add a copyright notice also at the top of each file, it highlights your deserved credit :) )
